validation/util/test.go

```diff
@@ -60,10 +62,10 @@ var lifecycleStatusMap = map[string]LifecycleStatus{
 // 2. Four phases for user to add his/her own operations.
 type LifecycleConfig struct {
 	Actions    LifecycleAction
-	PreCreate  func(runtime *Runtime) error
-	PostCreate func(runtime *Runtime) error
-	PreDelete  func(runtime *Runtime) error
-	PostDelete func(runtime *Runtime) error
+	PreCreate  func(runtime *Runtime) ([]byte, error)
```

Comment:
I'd rather not assume that these functions will include something that has a stderr.  Can we use [`*ExitError`][1] instead, and have the test include that stderr in their YAML diagnostic if the error casts to `*ExitError`?  I'll work up a PR with that narrow pivot to help show what I'm suggesting.

[1]: https://golang.org/pkg/os/exec/#ExitError

Reply:
> I'll work up a PR with that narrow pivot to help show what I'm suggesting.

Filed as #564.